### PR TITLE
handle interrupts, quote command properly on local side

### PR DESCRIPTION
**Pull request by keymone**
ssh with pseudo-tty and kill tasks on term/int/hup signals

### CODE REVIEW

**Comment by keymone** [comment]
this is PAASTA-10234

File: paasta_tools/cli/utils.py [hunk 1]
```diff
@@ -601,7 +601,7 @@ def run_on_master(cluster, system_paasta_config, cmd_parts,
     except NoMasterError as e:
         return (err_code, str(e))
 
-    ssh_parts = ['ssh', '-A', '-n', master]
+    ssh_parts = ['ssh', '-t', '-t', '-A', '-n', master]
```
Comment (solarkennedy) [inline paasta_tools/cli/utils.py]: two '-t' ? why do we need to force a tty?

File: paasta_tools/cli/utils.py [hunk 2]
```diff
@@ -595,23 +596,39 @@ def run_on_master(cluster, system_paasta_config, cmd_parts,
...
     try:
         master = connectable_master(cluster, system_paasta_config)
     except NoMasterError as e:
         return (err_code, str(e))
 
-    ssh_parts = ['ssh', '-A', '-n', master]
-
-    if shell:
-        ssh_parts.extend('sh -c "%s"' % quote(' '.join(cmd_parts)))
+    import subprocess
```
Comment (Rob-Johnson) [inline paasta_tools/cli/utils.py]: move this to the top of the file.

File: paasta_tools/cli/utils.py (hunk 1 above)
Comment (keymone) [inline paasta_tools/cli/utils.py]: i'm not entirely sure what's happening, but if i don't force tty stdout of cmd behind ssh stops flushing(?)

**Pull request merged**